xtevent/_eventiv.ado

```diff
@@ -284,11 +295,16 @@ program define _eventiv, rclass
 	

```

Comment:
@Constantino-Carreto-Romero  Do we need a check for whether the proxyiv lags to be normalized are in the estimation window if window was chosen to be max or balanced?

Reply:
@jorpppp 
For IV estimation, `_eventiv` makes two calls to `_eventgenvars`. The first call is to bring the imputed policyvar because it uses it to choose the lead order for the instrument. `_eventiv` also benefits from this first call and brings the calculated left window in https://github.com/JMSLab/xtevent/blob/267cef77ee4a32a1c26cce9d41e129d88cd45be0/xtevent/_eventiv.ado#L80. Then, in https://github.com/JMSLab/xtevent/blob/267cef77ee4a32a1c26cce9d41e129d88cd45be0/xtevent/_eventiv.ado#L86 it uses that value to define the limit for the lead orders to check in the code that selects the lead order later on in https://github.com/JMSLab/xtevent/blob/267cef77ee4a32a1c26cce9d41e129d88cd45be0/xtevent/_eventiv.ado#L135.

Reply:
@Constantino-Carreto-Romero So, if the user chose `window(max)` then first the window will be determined, and then the lead order will be chosen in that window. Sounds good.

However, my comment above was for the case when the user selects the leads, e.g. `window(max) proxyiv(2 3)`. Is that case handled by the second call to `event_iv`, such that it will check if the leads are in the chose window?

Reply:
@jorpppp 
There didn't exist such a check, even for the numeric window case. In https://github.com/JMSLab/xtevent/commit/74fa61b1888eb9165349e2ff7f69ba0b766623a6 I implemented such a check which works for both the numeric-defined window or the max/balanced case. 
For instance, when trying to run:
```
use example31, clear
cap noi xtevent y eta, panelvar(i) timevar(t) policyvar(z) proxy(x) impute(stag) window(max) proxyiv(5 e 20)

```
shows the following [error message](https://github.com/JMSLab/xtevent/assets/36095845/ca6560fe-fd37-4e58-8acd-404ce10ee86e).

Reply:
@Constantino-Carreto-Romero looks good, thanks.